utt/tests/testUtils.hpp

```diff
@@ -40,13 +45,13 @@ std::vector<uint32_t> getSubGroup(uint32_t n, uint32_t size) {
   return rret;
 }
 std::tuple<libutt::api::GlobalParams, RandSigDKG, RegAuthSK> init(size_t n, size_t thresh) {
-  GlobalParams d;
-  d.init();
-
+  GlobalParams::BaseLibsInitData base_libs_init_data;
+  GlobalParams::initLibs(base_libs_init_data);
   auto dkg = RandSigDKG(thresh, n, Params::NumMessages);
-  d.getParams() = Params::random(dkg.getCK());
-  auto rc = RegAuthSK::generateKeyAndShares(d.getParams().ck_reg, thresh, n, d.getParams().ibe);
-  d.getParams().ck_reg = rc.ck_reg;
+  auto rc = RegAuthSK::generateKeyAndShares(thresh, n);
+  GpData gp_data{dkg.getCK(), rc.ck_reg};
+  GlobalParams d = GlobalParams::create((void*)(&gp_data));
+  rc.setIBEParams(d.getParams().ibe);
```

Comment:
I don't think this is a "clean" interface to create a UTT instance. We should probably think how to improve it.
1) Are global params really global params? If they are - why aren't we using a singleton? if they're not - why are we calling them global params?
2) RandSigDKG gives no indication that we're creating a "bank authority", for "rc" it's almost the same, although you can probably make a guess from "RegAuthSK".
3) Creation is fragmented into little steps instead of a straightforward higher-level creation step.
4) Creation of global params is unsafe because it erases the type of the argument.
5) We start by creating "rc" but then we circle back to set some addition params for IBE that we need from global params - this is complicated to follow.

Reply:
I agree this could be improved, but:
1. Global params are per UTT instance, this is why we can't use a singleton
2. I agree that we can have a better naming, but I prefer not to change the legacy code naming for now
3. I think the higher level creation needs to be in the client library, I want to let the flexibility and the ability to control every step
4. See above
5. right, but I don't think we have any other option. To create the global params we need the registration commitment key (hence we first create rc) but the IBE params are part of the global params.
We can break it into even smaller pieces and initialize the IBE params separately, but I think it would be even more complicated